Ka-Block.safariextension/filter.js

```diff
@@ -3,20 +3,35 @@ function escape(domain) {
   return domain.replace(/[.]/g, '\\.')
 }
 
+// Get trigger url for block list rule
+function ruleURL(rule) {
+  return rule.trigger['url-filter']
+}
+
 // Create filter predicate function for blocked domains.
-function filter(domains) {
-  const all = domains.map(escape).join('|')
+function filter(rules) {
+  const all = rules.map(ruleURL).map(escape).join('|')
   const re = new RegExp('(?:^|\\.)(?:' + all + ')$', 'i')
   return re.test.bind(re)
 }
 
-// Predicate returns true for blocked domains.
-const blocked = filter(window.domains)
-
 // Respond to approval requests from start script.
-safari.application.addEventListener('message', function(event) {
-  if (event.name === 'canLoad') {
-    const host = new URL(event.message).hostname
-    event.message = blocked(host) ? 'block' : 'allow'
-  }
-}, true)
+function installCanLoad(rules) {
+  const blocked = filter(rules)
+  safari.application.addEventListener('message', function(event) {
+    if (event.name === 'canLoad') {
+      const host = new URL(event.message).hostname
+      event.message = blocked(host) ? 'block' : 'allow'
+    }
+  }, true)
+}
+
+function init() {
+  const req = new XMLHttpRequest()
+  req.responseType = 'json'
+  req.onload = function() { installCanLoad(req.response) }
+  req.open('GET', 'blockerList.json', true)
+  req.send()
```

Comment:
I wonder if loading the file asynchronously from the global page can return *after* the `start.js` code initializes on the current page. The global page JS runs only once at startup, so the only scenario in which this could occur is setting a homepage that needs ad-blocking and launching the app.

I tried this several times, and the JSON file load always finishes first. So, let's not worry about it for now.